services/httpd/service.go

```diff
@@ -165,3 +200,11 @@ func (s *Service) serve() {
 		s.err <- fmt.Errorf("listener failed: addr=%s, err=%s", s.Addr(), err)
 	}
 }
+
+// serve serves the handler from the unix socket listener.
```

Comment:
Can you also factor out the `serve()` method to accept a listener? That way you can do something like this:

```
func (s *Service) serve(ln net.Listener) {
    // above contents
}

func (s *Service) serveTCP() {
    s.serve(s.ln)
}

func (s *Service) serveUnixSocket() {
    s.serve(s.unixSocketListener)
}
```